File: MealMind.Api/Models/Recipe.cs
using System.ComponentModel.DataAnnotations;

namespace MealMind.Api.Models
{
    public class Recipe
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;
        public string? ImageUrl { get; set; }
        public NutritionInfo? Nutrition { get; set; }
        public string UserId { get; set; } = string.Empty;
    }
}

File: mealmind.web/src/types/recipe.ts
export interface NutritionInfo {
  calories: number;
  proteinGrams: number;
  carbsGrams: number;
  fatGrams: number;
}

export interface Ingredient {
  name: string;
  quantity: number;
  unit: string;
}

export interface Recipe {
  id: number;
  name: string;
  imageUrl: string | null;
  nutrition: NutritionInfo;
  steps: string[];
  portions: number;
  ingredients: Ingredient[];
  userId: string;
}
